imap/mailbox.c

```diff
@@ -5738,6 +5738,10 @@ static int _mailbox_index_repack(struct mailbox *mailbox,
                 repack->crcs.basic ^= crc_basic(&repack->newmailbox, &copyrecord);
                 repack->crcs.annot ^= crc_virtannot(&repack->newmailbox, &oldrecord);
                 repack->crcs.annot ^= crc_virtannot(&repack->newmailbox, &copyrecord);
+                repack->newmailbox.i.synccrcs.basic ^= crc_basic(&repack->newmailbox, &oldrecord);
+                repack->newmailbox.i.synccrcs.basic ^= crc_basic(&repack->newmailbox, &copyrecord);
+                repack->newmailbox.i.synccrcs.annot ^= crc_virtannot(&repack->newmailbox, &oldrecord);
+                repack->newmailbox.i.synccrcs.annot ^= crc_virtannot(&repack->newmailbox, &copyrecord);
             }
```

Comment:
Do we actually want to calculate all 4 of these CRCs twice, or should we assign them to a variable and then XOR that value against each of the two destination values?

Reply:
Yeah, we could do that!  In fact, we only need to calculate the CRC at all if that part has changed, so I've made it more efficient for every case.